apps/faucet/src/utils/addTokenToMetamask.ts

```diff
@@ -0,0 +1,51 @@
+import { TokenInput } from '../types';
+
+const getProvider = () => {
+  if (
+    typeof window !== 'undefined' &&
+    typeof (window as any).web3 !== 'undefined'
+  ) {
+    const provider =
+      (window as any).ethereum || (window as any).web3.currentProvider;
+
+    if (provider) {
+      return provider;
+    }
+  }
+};
+
+const addTokenToMetamask = async (tokenInput: TokenInput) => {
+  const { address, decimals, image, symbol } = tokenInput;
+
+  if (!address || !symbol) {
+    alert('Missing token address or symbol');
+    return;
+  }
+
+  const provider = getProvider();
+
+  if (provider) {
+    try {
+      await provider.request({
+        method: 'wallet_watchAsset',
+        params: {
+          options: {
+            address,
+            decimals,
+            image,
+            symbol,
+          },
+          type: 'ERC20',
+        },
+      });
+
+      console.log('Token added to Metamask');
+    } catch (error) {
+      console.error(error);
```

Comment:
If the token is added to Metamask already, the `provider.request` will throw an error or warning I'm pretty sure but we should somehow notify the user about that, we could do that in a separate PR.